### PR TITLE
Fix route selection regression and annotation selection

### PR DESCRIPTION
**Pull request by jill-cardamon**
### Description

This PR is to address a regression introduced in [#2901](https://github.com/mapbox/mapbox-navigation-ios/pull/2901) as well as fix MapView annotation selection.

### Implementation

Added a target to the existing tap gesture recognizer to address route and waypoint selection. Maps team also investigated the gesture recognizer conflict and found that the logic that causes MapView's single tap gesture recognizer to fail is not implemented yet. Tested on both the Nav SDK side and in 1Tap.

cc/ @bamx23

### CODE REVIEW

File: Sources/MapboxNavigation/NavigationMapView.swift [hunk 1]
```diff
@@ -297,10 +297,9 @@ open class NavigationMapView: UIView {
     }
     
     func setupGestureRecognizers() {
-        let gestures = gestureRecognizers ?? []
-        let mapTapGesture = UITapGestureRecognizer(target: self, action: #selector(didRecieveTap(sender:)))
-        mapTapGesture.requireFailure(of: gestures)
-        mapView.addGestureRecognizer(mapTapGesture)
+        for recognizer in mapView.gestureRecognizers ?? [] where recognizer is UITapGestureRecognizer {
```
Comment (bamx23) [inline Sources/MapboxNavigation/NavigationMapView.swift:300]: Do we need to add a target to all of the `UITapGestureRecognizer` instances?

Comment (jill-cardamon) [reply Sources/MapboxNavigation/NavigationMapView.swift:300]: We technically don't, but the gesture recognizers from the Maps SDK aren't named, so there's no way to filter through them effectively. The only way would be through target actions but those are private instance variables. That approach would either be fragile or rejected by Apple's review process. Trying to add to a certain gesture recognizer by order (ex// trying to add target to the last gesture recognizer in the `mapView.gestureRecognizer` array) also isn't very reliable. This fix is potentially temporary for now anyway until the Maps team implements logic on their side.

Comment (MaximAlien) [reply Sources/MapboxNavigation/NavigationMapView.swift:312]: @jill-cardamon, can you please create a follow-up ticket for this? If this is temporary workaround we should revise current implementation. If I understand this correctly idea would be to register only one gesture recognizer on Navigation SDK side, which will allow to select routes and waypoints. Such gesture recognizer is supposed to fail when such mechanism is available on Maps SDK side, right? 

Please also squash current PR into one commit, we do not want duplication in history based on previous PRs.

**Review by MaximAlien — APPROVED** [review]

File: Sources/MapboxNavigation/NavigationMapView.swift (hunk 1 above)
Comment (MaximAlien) [reply Sources/MapboxNavigation/NavigationMapView.swift:312]: It'd be also good to make sure that route selection logic works on Navigation SDK `1.x` and when custom gestures are added on client side.

**Pull request merged**